packages/@aws-cdk/aws-appmesh/lib/virtual-gateway-listener.ts

```diff
@@ -153,10 +153,27 @@ class VirtualGatewayListenerImpl extends VirtualGatewayListener {
  * Renders the TLS config for a listener
  */
 function renderTls(scope: Construct, tls: TlsListener | undefined): CfnVirtualGateway.VirtualGatewayListenerTlsProperty | undefined {
+  const trustProperty = tls?.validation?.trust.bind(scope).tlsValidationTrust;
+  if (trustProperty?.acm) {
```

Comment:
Can you explain why is ACM rejected everywhere? What's the point of having a `TlsValidationTrust` variant that can't be used anywhere?

Reply:
Previous to this code change, `TlsValidationTrust` was only used by `TlsClientPolicy` object. Similarly, `TlsCertificate` was only used by `TlsListener'. For these use cases, ACM can be used selected as well in addition to File and SDS options.

This code change allows `TlsValidationTrust` to be used by `TlsListener' and similarly `TlsCertificate` to be used by `TlsClientPolicy`. However, for these use cases, only File and SDS options can be selected.

Reply:
Thanks for the explanation @Seiya6329. Super helpful!

One more clarifying question. Is the limitation that ACM Certificates and Trust cannot be used for mutual TLS a fundamental one in AppMesh, or this is just a case of this not being implemented today, and this restriction is planned to be lifted in the (near) future?

Reply:
I found this [issue](https://github.com/aws/aws-app-mesh-roadmap/issues/258) and based from it, ACM is just not being implemented today. In addition, there seems be a plan to support ACM but no concrete target date is given.

Reply:
Thanks! In that case, let's go with my idea, described here: https://github.com/aws/aws-cdk/pull/15101#discussion_r653128220, to make passing ACM to mutual TLS impossible at compile-time.